### PR TITLE
Bump daisyui to v3 & update tailwindcss

### LINKED ISSUE #467: Upgrade DaisyUI to 3.x
We are using DaisyUI 2.x. 

3.x is been the official version for a few weeks. I think it's good if we keep it updated, so people won't get confused when reading the docs.

Changelog (3.0.0): https://daisyui.com/docs/changelog/#300-2023-06-01
Breaking changes: https://daisyui.com/docs/changelog/#breaking-changes

We can also update Tailwind (minor upgrade, still in v3)

### PR DESCRIPTION
**Pull request by technophile-04**
## Description

Migrated to daisy UI to v3 and minor update for tailwindcss 

Lol although the [changelog](https://daisyui.com/docs/changelog/#breaking-changes) was pretty straightforward but there were many internal things which needed to handle like `btn` class now adds `gap-2`  without you mentioning it so had to update elements which had two childs with parent as `btn` and many others small caveats like this in `dropdown` too

Nevertheless I went through all the changes properly so that all the things alighn with `main` branch 🙌

Also solved #366 

## Additional Information

- [X] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [X] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

Closes #467 , #122 

Your ENS/address: shivbhonde.eth

### CODE REVIEW

**Review by rin-st — APPROVED** [review]

**Comment by rin-st** [comment]
Works great!

File: packages/nextjs/components/example-ui/ContractInteraction.tsx [hunk 1]
```diff
@@ -63,15 +63,16 @@ export const ContractInteraction = () => {
...
                   onClick={() => writeAsync()}
+                  disabled={isLoading}
                 >
-                  {!isLoading && (
+                  {!isLoading ? (
                     <>
                       Send <ArrowSmallRightIcon className="w-3 h-3 mt-0.5" />
                     </>
+                  ) : (
+                    <span className="loading loading-spinner loading-sm"></span>
```
Comment (damianmarti) [inline packages/nextjs/components/example-ui/ContractInteraction.tsx:75]: Oh, I see, now you are showing this. OK.

File: packages/nextjs/components/scaffold-eth/Contract/DisplayVariable.tsx [hunk 2]
```diff
@@ -37,8 +37,12 @@ export const DisplayVariable = ({ contractAddress, abiFunction, refreshDisplayVa
     <div className="space-y-1 pb-2">
       <div className="flex items-center gap-2">
         <h3 className="font-medium text-lg mb-0 break-all">{abiFunction.name}</h3>
-        <button className={`btn btn-ghost btn-xs ${isFetching ? "loading" : ""}`} onClick={async () => await refetch()}>
-          {!isFetching && <ArrowPathIcon className="h-3 w-3 cursor-pointer" aria-hidden="true" />}
+        <button className="btn btn-ghost btn-xs" onClick={async () => await refetch()}>
+          {!isFetching ? (
```
Comment (damianmarti) [inline packages/nextjs/components/scaffold-eth/Contract/DisplayVariable.tsx:41]: Same, ask for isFetching

**Review by Pabl0cks — APPROVED** [review]
Working great @technophile-04 !! 🙌

Just one comment, should we add some padding to the `SearchBar.tsx` component in mobile?

search-bar-without-padding

File: packages/nextjs/components/example-ui/ContractInteraction.tsx (hunk 1 above)
Comment (technophile-04) [reply packages/nextjs/components/example-ui/ContractInteraction.tsx]: Yeah actually they have removed using `loading` class directly and suggest to this now checkout [here](https://daisyui.com/components/button/#button-with-loading-spinner) 🙌

File: packages/nextjs/components/example-ui/ContractInteraction.tsx [hunk 3]
```diff
@@ -63,15 +63,16 @@ export const ContractInteraction = () => {
...
                 <button
-                  className={`btn btn-primary rounded-full capitalize font-normal font-white w-24 flex items-center gap-1 hover:gap-2 transition-all tracking-widest ${
-                    isLoading ? "loading" : ""
-                  }`}
+                  className="btn btn-primary rounded-full capitalize font-normal font-white w-24 flex items-center gap-1 hover:gap-2 transition-all tracking-widest"
                   onClick={() => writeAsync()}
+                  disabled={isLoading}
                 >
-                  {!isLoading && (
+                  {!isLoading ? (
```
[the remark this answers is not in the document]
Comment (technophile-04) [reply packages/nextjs/components/example-ui/ContractInteraction.tsx]: Yes ! lol I had this mind but forgot in the end, Fixed at [003da20](https://github.com/scaffold-eth/scaffold-eth-2/pull/490/commits/003da2092f8e5a0f27c21a09e020d2331c164d03) Thanks 🙌

**Comment by technophile-04** [comment]
> Just one comment, should we add some padding to the SearchBar.tsx component in mobile?

Fixed at [6192187](https://github.com/scaffold-eth/scaffold-eth-2/pull/490/commits/619218796fdedee9707737c7240b2af5381f37f7) 🙌

Screenshot 2023-08-16 at 1 48 59 PM


Thanks all for testing and reviews merging this 🙌

**Pull request merged**